onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py

```diff
@@ -313,12 +314,103 @@ def optimize_onnx(
 
         m.save_model_to_file(optimized_model_path, use_external_data_format, all_tensors_to_one_file=True)
 
+    @staticmethod
+    def pt_transcription_for_verify_onnx(
+        ds: Union[datasets.DatasetDict, datasets.Dataset, datasets.IterableDatasetDict, datasets.IterableDataset],
+        processor: WhisperProcessor,
+        pt_model: torch.nn.Module,
+        device: torch.device,
+        batch_size: int = 1,
+        prompt_mode: bool = False,
+    ):
+        input_features_ = []
+        if batch_size == 1:
+            input_features = processor([ds[0]["audio"]["array"]], return_tensors="pt").input_features
+        else:
+            input_features_ = [
+                processor([ds[3]["audio"]["array"]], return_tensors="pt").input_features,
+                processor([ds[3]["audio"]["array"]], return_tensors="pt").input_features,
+            ]
+            assert len(input_features_) == batch_size
+            input_features = torch.cat((input_features_[0], input_features_[1])).to(device)
```

Comment:
```suggestion
            input_features = torch.cat((input_features_[0], input_features_[1]))
```
The input features appear to be moved to the device in both cases.

https://github.com/microsoft/onnxruntime/blob/b34a04af6f79598ee8968c5ac0d82155c55de7ce/onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py#L339-L340